packages/clerk-js/src/ui.retheme/primitives/hooks/useFormControl.tsx

```diff
@@ -1,111 +1,41 @@
-   * - The form element (e.g. Input) gets the `id`
-   */
-  id: string;
-  isRequired?: boolean;
-  hasError?: boolean;
-  isDisabled?: boolean;
-  setError: (error: string | ClerkAPIError | undefined) => void;
-  setSuccess: (message: string) => void;
-  setWarning: (warning: string) => void;
-  setInfo: (info: string) => void;
-  clearFeedback: () => void;
-};
-
-/**
- * @deprecated
- */
-type FormControlContextValue = Required<FormControlProps> & { errorMessageId: string };
-
-/**
- * @deprecated Use FormFieldContextProvider
- */
-export const [FormControlContext, , useFormControl] =
-  createContextAndHook<FormControlContextValue>('FormControlContext');
-
-/**
- * @deprecated Use FormFieldContextProvider
- */
-export const FormControlContextProvider = (props: React.PropsWithChildren<FormControlProps>) => {
-  const {
-    id: propsId,
-    isRequired = false,
-    hasError = false,
-    isDisabled = false,
-    setError,
-    setSuccess,
-    setWarning,
-    setInfo,
-    clearFeedback,
-  } = props;
-  // TODO: This shouldnt be targettable
-  const id = `${propsId}-field`;
-  /**
-   * Track whether the `FormErrorText` has been rendered.
-   * We use this to append its id the `aria-describedby` of the `input`.
-   */
-  const errorMessageId = hasError ? `error-${propsId}` : '';
-  const value = React.useMemo(
-    () => ({
-      value: {
-        isRequired,
-        hasError,
-        id,
-        errorMessageId,
-        isDisabled,
-        setError,
-        setSuccess,
-        setWarning,
-        setInfo,
-        clearFeedback,
-      },
-    }),
-    [isRequired, hasError, id, errorMessageId, isDisabled, setError, setSuccess, setInfo, setWarning, clearFeedback],
-  );
-  return <FormControlContext.Provider value={value}>{props.children}</FormControlContext.Provider>;
-};
-
 type FormFieldProviderProps = ReturnType<typeof useFormControlUtil<FieldId>>['props'] & {
-  hasError?: boolean;
-  isDisabled?: boolean;
+  isDisabled: boolean;
 };
 
 type FormFieldContextValue = Omit<FormFieldProviderProps, 'id'> & {
   errorMessageId?: string;
   id?: string;
   fieldId?: FieldId;
+  hasError: boolean;
 };
-export const [FormFieldContext, useFormField] = createContextAndHook<FormFieldContextValue>('FormFieldContext');
+export const [FormFieldContext, , useFormField] = createContextAndHook<FormFieldContextValue>('FormFieldContext');
 
 export const FormFieldContextProvider = (props: React.PropsWithChildren<FormFieldProviderProps>) => {
   const {
     id: propsId,
     isRequired = false,
     isDisabled = false,
-    hasError = false,
     setError,
     setSuccess,
     setWarning,
     setHasPassedComplexity,
     setInfo,
     clearFeedback,
     children,
+    feedbackType,
     ...rest
   } = props;
+  // The following TODO existed beforehand, it is simply copied during the refactor
   // TODO: This shouldnt be targettable
   const id = `${propsId}-field`;
```

Comment:
The following TODO existed beforehand, it is simply copied during the refactor

Reply:
🙃 should we remove the comment since this is only intended to be read in the PR review and it does not give more context on the code?

Reply:
could u recheck, i think i removed it